src/expression.d

```diff
@@ -1210,11 +1208,14 @@ extern (C++) bool arrayExpressionToCommonType(Scope* sc, Expressions* exps, Type
  */
 extern (C++) TemplateDeclaration getFuncTemplateDecl(Dsymbol s)
 {
-    FuncDeclaration f = s.isFuncDeclaration();
+    auto f = s.isFuncDeclaration();
```

Comment:
When we use downcast method, the target type name is already in rhs - is"FuncDeclaration". So this change happily reduces redundancy.